### ISSUE TITLE
ScrapeHandler.Process(...) does not seem to return metric output in a deterministic sequence

### ISSUE THREAD

**Issue by davidalpert** [PrometheusClientNet/Prometheus.Client#59]
Is there expected to be any deterministic sort order with the scraped metrics when using:
```
        ScrapeHandler.Process(registry, memoryStream);
```

it appears that related metrics (i.e. with similar names) do not sort together and I may have seen the sort order change from scrape to scrape.

is this expected? 

I will work on a PR with a unit test demonstrating this.

**Comment by sanych-sun** [comment]
It's a good question. As far as I understood there is no technical needs to sort them, as Prometheus server ignores sequence. But for debug purposes it could be usefull.

Are you using v2 of the library? We did a lot of changes in v3 and described behavior could change in v3. Could you please check release candidate package of v3 build?

**Comment by davidalpert** [comment]
Thanks for the quick response! 

I'm currently using version `2.2.1` and I should have included that in my question, thanks for the reminder.

I agree that Prometheus server doesn't seem to care about the sort order but sorting the scraped metrics alphabetically would ensure that metrics with related names are grouped in a natural and predictable sequence when scanned by a human.

I will try updating to v3 and if it's still not sorting for me in a deterministic way, then I can submit a PR, knowing that this is:

1. a known issue; and
1. something which would not impact functionality but may improve user experience.

**Comment by sanych-sun** [comment]
Sure! Please let us know if you need some more information, help or additional changes. It's a good time to jump into v3 release ;-) .

**Comment by davidalpert** [comment]
v3 release is listed as `prerelease` - is it stable?

**Comment by sanych-sun** [comment]
It's release candidate. If no issues appears we will make it release in near future.

**Comment by davidalpert** [comment]
In build `3.0.0-rc2` the `CollectorRegistry` class has no more `Instance` property; is there a new pattern recommended for scraping results in a controller action?

**Comment by sanych-sun** [comment]
It's available via Metrics.DefaultCollectorRegistry

**Comment by sanych-sun** [comment]
@davidalpert , I've checked v3 and found that metrics returned in the historical sequence (first registered appears first in scrap) and sequence seems to be constant. But personally I like your suggestion to sort them by name, so I've created PR to sort this out.

**Comment by phnx47** [comment]
How performance changes?

If productivity decreases, it is better to create a debug mode: Sort alphabetically, add '\n' between metrics and etc.

**Comment by phnx47** [comment]
Why not debug in Prometheus(not client)?

**Comment by sanych-sun** [comment]
Memory allocation the same. Performance impact is minimal, because actual sorting happens once after collectors modifications (add or remove metrics) but before scrape, sorted collection persist between scrapes unless there were changes to collectors registry. So if client code registers all metrics on start up (which is recommended way to do), the sorting occurs only once.

**Comment by sanych-sun** [comment]
@davidalpert the changes were merged into master, you can try it via the my-get package, or wait for next rc or release package.
